modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -833,8 +834,51 @@ def raise_exception_if_value_not_present(val, form_obj_description)
         )
       end
 
+      def validate_claim_process_type_bdd!
+        claim_process_type = form_attributes['claimProcessType']
+        if claim_process_type == 'BDD_PROGRAM'
+          date = form_attributes['claimDate'] || Time.find_zone!('Central Time (US & Canada)').today
+          claim_date = Date.parse(date.to_s)
+          service_information = form_attributes['serviceInformation']
+          title_ten_activation = service_information&.dig('federalActivation')
+          anticipated_seperation_date = title_ten_activation&.dig('anticipatedSeparationDate')
+
+          unless active_duty_end_date_qualifies(service_information, claim_date) ||
+                 anticipated_separation_date_qualifies(anticipated_seperation_date, claim_date)
+            raise ::Common::Exceptions::UnprocessableEntity.new(
+              detail: 'Must have an activeDutyEndDate or anticipatedSeparationDate between 90 & 180 days' \
+                      ' from claim date.'
+            )
+          end
+        end
+      end
+
       private
 
+      # Confirm at least one activeDutyEndDate in BDD range
+      def active_duty_end_date_qualifies(service_information, claim_date)
+        service_information['servicePeriods'].each do |sp|
+          if sp['activeDutyEndDate']
+            active_duty_end_date = Date.strptime(sp['activeDutyEndDate'], '%m-%d-%Y')
+            return true if date_in_bdd_claim_date_range(active_duty_end_date, claim_date)
+          end
+        end
+        false
+      end
+
+      # Confirm anticipatedSeparationDate in BDD range
+      def anticipated_separation_date_qualifies(anticipated_seperation_date, claim_date)
+        if anticipated_seperation_date
+          date_in_bdd_claim_date_range(Date.strptime(anticipated_seperation_date, '%m-%d-%Y'), claim_date)
+        else
+          false
+        end
+      end
+
+      def date_in_bdd_claim_date_range(target_date, claim_date)
+        target_date.between?(claim_date.next_day(90), claim_date.next_day(180))
+      end
+
```

Comment:
I think the entirety of changes in this file could roughly translate to something a bit simpler.
```
      def validate_claim_process_type_bdd!
        claim_process_type = form_attributes['claimProcessType']
        if claim_process_type == 'BDD_PROGRAM'
          claim_date = Date.parse(form_attributes['claimDate']) || Time.find_zone!('Central Time (US & Canada)').today
          service_information = form_attributes['serviceInformation']
          active_dates = service_information['servicePeriods']&.pluck('activeDutyEndDate')
          active_dates << service_information&.dig('federalActivation', 'anticipatedSeparationDate')

          unless active_dates.compact.map do |a|
                   Date.strptime(a, '%m-%d-%Y').between?(claim_date.next_day(90), claim_date.next_day(180))
                 end.any?
            raise ::Common::Exceptions::UnprocessableEntity.new(
              detail: 'Must have an activeDutyEndDate or anticipatedSeparationDate between 90 & 180 days' \
                      ' from claim date.'
            )
          end
        end
      end
```

but this is just a rough take at it with the example I have. I have not thoroughly tested. I feel like we can get away with moving this to just plucking the dates, iterating through and checking for any between the required dates. Seems like we're passing specific arrays in and have multiple functions to handle specific formats where we could just pass dates to one consistent method. 

If we want to go real crazy we could write constants for 90 and 180 so they're easy to change. If we update the constant we get it in the `next_day(BDD_LOWER_LIMIT)` and the error with something like `between #{BDD_UPPER_LIMIT}`.

The date manipulation in the unless line I think can be removed when we go back to (wait for it...... ) ISO dates. Sorry, I had to 😆 



I would almost take it one step further and add a method for the first line and the if, and only call line 35 if it's true.. so something like  `validate_claim_process_type_bdd! if bdd_claim?`